### PR TITLE
add SSL certificate configuration support for HTTP/3

### PR DESCRIPTION
**Pull request by blueuuufish**
## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Review by AlbumenJ — COMMENTED** [review]
@oxsean PTAL

File: dubbo-rpc/dubbo-rpc-triple/src/test/java/org/apache/dubbo/rpc/protocol/tri/TripleHttp3ProtocolTest.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        java.net.URL clientCertUrl = TripleHttp3ProtocolTest.class.getResource("/certs/client.pem");
+        java.net.URL clientKeyUrl = TripleHttp3ProtocolTest.class.getResource("/certs/client.key");
+        java.net.URL trustCertUrl = TripleHttp3ProtocolTest.class.getResource("/certs/ca.pem");
+        Assertions.assertNotNull(serverCertUrl, "serverCertUrl should not be null");
+        sslConfig.setServerKeyCertChainPath(
+                Paths.get(serverCertUrl.toURI()).toAbsolutePath().toString());
+        Assertions.assertNotNull(serverKeyUrl, "serverKeyUrl should not be null");
+        sslConfig.setServerPrivateKeyPath(
+                Paths.get(serverKeyUrl.toURI()).toAbsolutePath().toString());
+        Assertions.assertNotNull(trustCertUrl, "serverKeyUrl should not be null");
```
Comment (oxsean) [inline dubbo-rpc/dubbo-rpc-triple/src/test/java/org/apache/dubbo/rpc/protocol/tri/TripleHttp3ProtocolTest.java]: trustCertUrl

File: dubbo-rpc/dubbo-rpc-triple/src/test/java/org/apache/dubbo/rpc/protocol/tri/TripleHttp3ProtocolTest.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        sslConfig.setServerTrustCertCollectionPath(
+                Paths.get(trustCertUrl.toURI()).toAbsolutePath().toString());
+        Assertions.assertNotNull(clientCertUrl, "clientCertUrl should not be null");
+        sslConfig.setClientKeyCertChainPath(
+                Paths.get(clientCertUrl.toURI()).toAbsolutePath().toString());
+        Assertions.assertNotNull(clientKeyUrl, "serverKeyUrl should not be null");
```
Comment (oxsean) [inline dubbo-rpc/dubbo-rpc-triple/src/test/java/org/apache/dubbo/rpc/protocol/tri/TripleHttp3ProtocolTest.java]: clientKeyUrl

File: dubbo-rpc/dubbo-rpc-triple/src/test/java/org/apache/dubbo/rpc/protocol/tri/TripleHttp3ProtocolTest.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        sslConfig.setClientPrivateKeyPath(
+                Paths.get(clientKeyUrl.toURI()).toAbsolutePath().toString());
+        Assertions.assertNotNull(trustCertUrl, "trustCertUrl should not be null");
+        sslConfig.setClientTrustCertCollectionPath(
```
Comment (oxsean) [inline dubbo-rpc/dubbo-rpc-triple/src/test/java/org/apache/dubbo/rpc/protocol/tri/TripleHttp3ProtocolTest.java]: You can create a common method such as:
```java
private static String getCertPath(String name) {
     java.net.URL certUrl = TripleHttp3ProtocolTest.class.getResource("/certs/"+name);
    Assertions.assertNotNull(certUrl, "Cert file '/certs/"+ name + "' is required");
    return Paths.get(certUrl.toURI()).toAbsolutePath().toString()
}
```

File: dubbo-rpc/dubbo-rpc-triple/src/test/resources/certs/client.key [hunk 4]
```diff
@@ -0,0 +1,28 @@
+-----BEGIN PRIVATE KEY-----
```
Comment (oxsean) [inline dubbo-rpc/dubbo-rpc-triple/src/test/resources/certs/client.key:1]: No need for csr

File: dubbo-rpc/dubbo-rpc-triple/src/test/resources/certs/server.key [hunk 5]
```diff
@@ -0,0 +1,28 @@
+-----BEGIN PRIVATE KEY-----
```
Comment (oxsean) [inline dubbo-rpc/dubbo-rpc-triple/src/test/resources/certs/server.key:1]: Confirm all certificates are valid for more than 50 years

**Review by oxsean — CHANGES_REQUESTED** [review]

File: dubbo-rpc/dubbo-rpc-triple/src/test/resources/certs/server.key (hunk 5 above)
Comment (blueuuufish) [reply dubbo-rpc/dubbo-rpc-triple/src/test/resources/certs/server.key:1]: Thank you for pointing that out.

File: dubbo-rpc/dubbo-rpc-triple/src/test/java/org/apache/dubbo/rpc/protocol/tri/TripleHttp3ProtocolTest.java (hunk 3 above)
Comment (blueuuufish) [reply dubbo-rpc/dubbo-rpc-triple/src/test/java/org/apache/dubbo/rpc/protocol/tri/TripleHttp3ProtocolTest.java]: Thank you for pointing that out.

File: dubbo-rpc/dubbo-rpc-triple/src/test/resources/certs/client.key (hunk 4 above)
Comment (blueuuufish) [reply dubbo-rpc/dubbo-rpc-triple/src/test/resources/certs/client.key:1]: Thank you, I've already corrected it.

**Review by oxsean — APPROVED** [review]

**Comment by oxsean** [comment]
@AlbumenJ LGTM.

**Pull request merged**